pkg/cli/internal/cmd/e2e-test/setup-e2e-operator.go

```diff
@@ -0,0 +1,175 @@
+}
+
+func newSetupE2EOperator(log logr.Logger) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "setup-e2e-operator",
+		Short: "install e2e operator in the given cluster",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			kubeconfig := cmd.Flag("kubeconfig").Value.String()
+			namespaceName := cmd.Flag("namespace").Value.String()
+			if err := setupE2EOperator(log, kubeconfig, namespaceName, cmd.OutOrStdout()); err != nil {
+				return fmt.Errorf("setup-e2e: %w", err)
+			}
+			return nil
+		},
+	}
+	cmd.Flags().String("kubeconfig", os.Getenv("KUBECONFIG"), "cluster kubeconfig where to install")
+	cmd.Flags().StringP("namespace", "n", "kubecarrier-e2e-operator", "namespace where to deploy operator")
+	return cmd
+}
+
+func setupE2EOperator(log logr.Logger, kubeconfig string, namespaceName string, output io.Writer) error {
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespaceName,
+		},
+	}
+	if kubeconfig != "" {
+		if err := os.Setenv("KUBECONFIG", kubeconfig); err != nil {
+			return fmt.Errorf("kubeconfig env setup: %w", err)
+		}
+	}
+	ctx := context.Background()
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return fmt.Errorf("getting Kubernetes cluster config: %w", err)
+	}
+
+	c, err := client.New(cfg, client.Options{Scheme: scheme})
+	if err != nil {
+		return fmt.Errorf("creating Kubernetes client: %w", err)
+	}
+	s := wow.New(output, spin.Get(spin.Dots), "spinner text")
+
+	startTime := time.Now()
+	if err := spinner.AttachSpinnerTo(s, startTime, "creating namespace", func() error {
+		_, err := controllerutil.CreateOrUpdate(ctx, c, namespace, func() error {
+			return nil
+		})
+		return err
+
+	}); err != nil {
+		return err
+	}
+
+	if err := spinner.AttachSpinnerTo(s, startTime, "deploy e2e-operator", func() error {
+		objects, err := fakeoperator.Manifests(
+			fakeoperator.Config{
+				Namespace: namespace.Name,
+			})
+		if err != nil {
+			return fmt.Errorf("creating operator manifests: %w", err)
+		}
+
+		for _, object := range objects {
+			if err := controllerutil.SetControllerReference(namespace, &object, scheme); err != nil {
+				return fmt.Errorf("set controller reference: %w", err)
+			}
+			b, err := yaml.Marshal(object)
+			if err != nil {
+				// this should never ever happen
+				panic(err)
+			}
+			log.V(9).Info("Creating object\n" + string(b))
+			_, err = reconcile.Unstructured(ctx, log, c, &object)
+			if err != nil {
+				return fmt.Errorf("reconcile kind: %s, err: %w", object.GroupVersionKind().Kind, err)
+			}
+			log.V(6).Info("reconciled",
+				"name", object.GetName(),
+				"namespace", object.GetNamespace(),
+				"kind", object.GroupVersionKind().Kind,
+				"group", object.GroupVersionKind().Group,
+				"version", object.GroupVersionKind().Version,
+			)
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if err := spinner.AttachSpinnerTo(s, startTime, "waiting for available deployment", func() error {
```

Comment:
Please remove this function and use the https://github.com/kubermatic/kubecarrier/blob/master/pkg/cli/internal/cmd/setup/setup.go#L147-L179 with the function above

Reply:
but this function for `kubecarrier`, I can reuse it